modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js

```diff
@@ -54,24 +53,12 @@ export const getLabel = (originalField, editingLanguageId) => {
 	]);
 };
 
-export const getValidation = (originalField, oldFieldName, newFieldName) => {
+export const getValidation = originalField => {
 	const validation = getSettingsContextProperty(
 		originalField.settingsContext,
 		'validation'
 	);
 
-	const expression = validation.expression;
-
-	if (expression && expression.value) {
-		return {
-			...validation,
-			expression: {
-				...validation.expression,
-				value: expression.value.replace(oldFieldName, newFieldName),
-			},
-		};
-	}
-
```

Comment:
@carolmariaabb, why this is no more necessary?

Reply:
It's not necessary because [fieldDuplicatedHandler.es.js#L29-L34](https://github.com/carolmariaabb/liferay-portal/blob/91ac14b75b10335551e4488654f74fb1223880b6/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js#L29-L34) already updates the _fieldname_ for the expression parameter. 
If you keep this code and the duplicated field is _CopyOfText_, the expression will become `isUrl(CopyOfCopyOfText)` instead of `isUrl(CopyOfText)`, for example.

Reply:
Nice, @carolmariaabb! Thanks for your clarification. 😉 

Once you removed the core of the method `getValidation`, does it is still necessary? Can we remove it?

Reply:
It's still necessary because the _json_ looks like this:

`{
	name: 'TextField',
	settingsContext: {
      pages: [
			{
				rows: [
					{
						columns: [
							{
								fields: [
									{
										fieldName: 'validation',
										value: {}
									},
								],
							},
						],
					},
				],
			},
		],
	},
    type: 'text',
    validation: {}
}`

[fieldDuplicatedHandler.es.js#L29-L34](https://github.com/carolmariaabb/liferay-portal/blob/91ac14b75b10335551e4488654f74fb1223880b6/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js#L29-L34)  only updates the _fieldname_ inside the `settingsContext` ([settingsContext.es.js#L86-L93](https://github.com/liferay/liferay-portal/blob/615dafa27f7ed72a5b17352cb7c60d0ebc432e29/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/util/settingsContext.es.js#L86-L93)).

Reply:
The `getValidation` it's necessary [here](https://github.com/carolmariaabb/liferay-portal/blob/91ac14b75b10335551e4488654f74fb1223880b6/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js#L38-L44) to update the validation property outside the `settingsContext`.

Reply:
Awesome! Thanks for your clarification, @carolmariaabb. 😄